tapdb/universe_stats.go

```diff
@@ -837,6 +840,13 @@ func (u *UniverseStats) QuerySyncStats(ctx context.Context,
 
 	// Finally, we'll create the time after function that'll wipe the
 	// cache, forcing a refresh.
+	//
+	// If we already have a timer active, then stop it, so we only have a
+	// single timer going at any given time.
+	if u.syncStatsRefresh != nil && !u.syncStatsRefresh.Stop() {
+		<-u.syncStatsRefresh.C
```

Comment:
what are the guarantees of reading immediately from `u.syncStatsRefresh.C`?
could also do
```suggestion
                select {
		case <-u.syncStatsRefresh.C:
		case <-ctx.Done():
			return nil, ctx.Err()
		}
```